### PR TITLE
test

### PR DESCRIPTION
**Pull request by hamedhsn**
SQLite: support a variant of NumericValue.summary #597

### CODE REVIEW

File: ibis/sql/compiler.py [hunk 1]
```diff
@@ -1254,8 +1254,16 @@ def _category_label(expr):
     return stmt.end().name(expr._name)
 
 
+@rewrites(ops.Sum)
```
Comment (hamedhsn) [inline ibis/sql/compiler.py]: @wesm I was trying to do a test to use rewrites decorator for SUM. why does it not get called?

Comment (wesm) [inline ibis/sql/compiler.py]: I'm not sure what you're testing, but you may need to look at the ExprTranslator class where the rewrite rules are applied

**Comment by hamedhsn** [comment]
@wesm need your help..so we would have class NumericSummary in operation and then rewrites in compiler to rewrite each operation in summary.so the actual implementation of each operation would be here. right? 
then how would be the NumericSummary implementation?

**Comment by wesm** [comment]
I won't be able to walk you through it, but perhaps look at the implementation of `NullIfZero` in `ibis/sql/alchemy.py` and the related operation / user API implementation. 

This in particular is a difficult task to be very honest -- there will probably be some refactoring required in the generic SQL compiler to make the result of `summary` behave in an intuitive way.

**Comment by hamedhsn** [comment]
thanks I am looking into it.  
if you come across any easy task let me know pls.

**Comment by wesm** [comment]
I thought some more about this and I think rewriting summary would cause too many problems / edge cases. It would be better to have a global (or thread local) flag (plus an option in the summary method) that indicates the flavor of backend so that not implemented aggregates can be just dropped. That is much easier

**Comment by hamedhsn** [comment]
Is that what you meant? I added an attribute at the time of the connect so that it is checked in summary methods to call/drop some operations.

**Comment by wesm** [comment]
No, that's not what I'm after. Let's leave this one and I'll keep an eye out for other issues to tackle

**Comment by hamedhsn** [comment]
ok. sorry.
Please let me know, thanks.

**Pull request closed without merging**